terror/terror.go

```diff
@@ -179,6 +183,15 @@ func (e *Error) Code() ErrCode {
 	return e.code
 }
 
+// SetWorkaround is a decorator like method which add a workaround to
+// error which is convenient for user to search.
+func (e *Error) SetWorkaround(workaround, description string) *Error {
```

Comment:
What does `description` mean? Could you give an example usage of this method?

Reply:
error messge is enough, I will remove it.

Reply:
Example is like:
`ErrUnknown = terror.ClassOptimizer.New(mysql.ErrUnknown, mysql.MySQLErrName[mysql.ErrUnknown]).SetWorkaround("here is workaround")`

Reply:
IMHO it's better to move setting `description` to another `SetDescription` just like `SetWorkaround`, so that developers can choose if they want to set the `description` explicitly.